### DISCUSSION TITLE
External macro implementation type 'CasePathsMacros.CasePathableMacro' could not be found for macro 'CasePathable()'

### DISCUSSION THREAD

**Discussion by Drag0ndust** [pointfreeco/swift-composable-architecture#2652] [Q&A]
I've initiated a new project and aimed to incorporate TCA version 1.5.5. 
However, when I created my initial feature, I encountered an error with the Action enum: "External macro implementation type 'CasePathsMacros.CasePathableMacro' could not be found for macro 'CasePathable()'."

here is the Feature Code:
```swift
@Reducer
struct ProjectsFeature: Reducer {
    struct State: Equatable {}

    enum Action {
        case test
    }

    var body: some ReducerOf<Self> {
        Reduce { state, action in
            switch action {
            case .test:
                return .none
            }
        }
    }
}
```

What might be the issue here?

**Comment by mbrandonw** [comment]
Hi @Drag0ndust, can you provide a project that demonstrates this behavior? I just created a fresh project with TCA and your code seems to work fine for me. Also what version of Xcode are you using?

**Reply by Drag0ndust** [reply]
I use Xcode 15.0, Swift 5.9

I just created a new plain project and used the internal SPM to use the Package. There it works.
In my other example I use Tuist (tuist.io) to generate the project files and there it isn't working.
It seems like the issue is on the tuist side.

I will recheck it...

**Reply by mbrandonw** [reply]
Hi @Drag0ndust, yes this has come up a few times from people. I'm not sure if tuist supports macros yet, or if you need to update your version of tuist.

**Reply by Drag0ndust** [reply]
I will try to solve this problem. When I have a solution, I will post it here

**Comment by roya1v** [comment]
I was having the same issue, fixed it using this sample from tuist https://github.com/tuist/tuist/tree/main/fixtures/framework_with_swift_macro

**Comment by maleem88** [comment]
hi @mbrandonw
faced the same error using SPM, but built with terminal using xcodebuild.
its compiling fine in xcode, but fail with this error in the terminal

**Reply by mbrandonw** [reply]
Hi @mohamed-aleem, please provide a minimal project that demonstrates the problem you are seeing. There is not enough information in your message to help.

**Reply by Alex293** [reply]
If you are passing a sdk parameter to Xcodebuild, drop it. It forces xcodebuild to build macro using the same sdk as the target instead as for the host.

**Reply by maleem88** [reply]
Brilliant, thanks @Alex293 this resolved the issue.

thanks Brandon @mbrandonw for your reply , I should have uploaded more details to make it clearer. 
btw, I use composable architecture for a quiet big app, and used xcodebuild command to detect compile bottlenecks, and found that XCTestDynamicOverlay "unimplemented()" method take significant time in compile time. I was able to cut the compile time by 75% (from 175s to 40s) by reverting back to plain functions for mocks. 

<img width="1089" alt="Screenshot 2024-01-13 at 7 15 17 AM" src="https://github.com/pointfreeco/swift-composable-architecture/assets/3467482/8ef37dc5-1292-41a1-85f3-840a780b9d55">

**Reply by Alex293** [reply]
I think this is a bug I’ll open a radar for it and you should too
